src/main/java/org/opensearch/knn/quantization/models/quantizationState/MultiBitScalarQuantizationState.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationState;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.opensearch.knn.quantization.models.quantizationParams.ScalarQuantizationParams;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+/**
+ * MultiBitScalarQuantizationState represents the state of multi-bit scalar quantization,
+ * including the thresholds used for quantization.
+ */
+@Getter
+@NoArgsConstructor // No-argument constructor for deserialization
+@AllArgsConstructor // Constructor with all arguments
+public final class MultiBitScalarQuantizationState implements QuantizationState {
+    private ScalarQuantizationParams quantizationParams;
+    /**
+     * The threshold values for multi-bit quantization, organized as a 2D array
+     * where each row corresponds to a different bit level.
+     *
+     * For example:
+     * - For 2-bit quantization:
+     *   thresholds[0] -> {0.5f, 1.5f, 2.5f}  // Thresholds for the first bit level
+     *   thresholds[1] -> {1.0f, 2.0f, 3.0f}  // Thresholds for the second bit level
+     * - For 4-bit quantization:
+     *   thresholds[0] -> {0.1f, 0.2f, 0.3f}
+     *   thresholds[1] -> {0.4f, 0.5f, 0.6f}
+     *   thresholds[2] -> {0.7f, 0.8f, 0.9f}
+     *   thresholds[3] -> {1.0f, 1.1f, 1.2f}
+     *
+     * Each column represents the threshold for a specific dimension in the vector space.
+     */
+    private float[][] thresholds;
+    private static final long serialVersionUID = 1L; // Version ID for serialization
+    private static final int CURRENT_VERSION = 1; // Current version of SQParams
+
+    @Override
+    public ScalarQuantizationParams getQuantizationParams() {
+        return quantizationParams;
+    }
+
+    @Override
+    public void writeExternal(ObjectOutput out) throws IOException {
+        int combinedVersion = (CURRENT_VERSION << 16) | quantizationParams.getVersion();
```

Comment:
Is`|` done to save memory. I am wondering why is this coupled and if there are any unforeseen consequences . Should  `quantizationParams.writeExternal(out)` be taking care of quantizationparams version?

Reply:
We are writing version once , and we don't want write version for every child classes and yes it's for saving memory

Reply:
How will the version be handled in case quantizationParams is written independently? Being a public isn't that a possibility?